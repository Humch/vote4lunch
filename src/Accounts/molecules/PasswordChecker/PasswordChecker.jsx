import React from 'react';
import { connect } from 'react-redux';

import PropTypes from 'prop-types';

import { ValueChecker } from '../../../Core';

const PasswordChecker = ({ password, passwordRepeater, tooShort }) => {
  let rendering = '';
  if (tooShort) {
    rendering = <ValueChecker check={false} />;
  } else if (password === passwordRepeater) {
    rendering = <ValueChecker check />;
  } else {
    rendering = <ValueChecker check={false} />;
  }
  return (
    <span>
      { rendering }
    </span>
  );
};

PasswordChecker.propTypes = {
  password: PropTypes.string.isRequired,
  passwordRepeater: PropTypes.string.isRequired,
  tooShort: PropTypes.bool.isRequired,
};

const mstp = ({ userData, passwordChecker }) => ({
  password: userData.password,
  passwordRepeater: userData.passwordRepeater,
  tooShort: passwordChecker.tooShort,
});

export default connect(mstp)(PasswordChecker);
