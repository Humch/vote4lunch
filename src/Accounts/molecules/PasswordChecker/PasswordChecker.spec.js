/* eslint-disable no-undef */
import React from 'react';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import PasswordChecker from './PasswordChecker';

describe('PasswordChecker Snapshot', () => {
  const initialState = {
    userData: {
      password: 'f4k3',
      passwordRepeater: 'f4k3',
    },
    passwordChecker: {
      tooShort: true,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of PasswordChecker with too short passwords', () => {
    const renderedValue = renderer.create(<PasswordChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('PasswordChecker Snapshot', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: '',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of PasswordChecker with passwords equals', () => {
    const renderedValue = renderer.create(<PasswordChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('PasswordChecker Snapshot', () => {
  const initialState = {
    userData: {
      password: '',
      passwordRepeater: 'f4k3',
    },
    passwordChecker: {
      tooShort: false,
    },
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('capturing Snapshot of PasswordChecker with different passwords', () => {
    const renderedValue = renderer.create(<PasswordChecker store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
