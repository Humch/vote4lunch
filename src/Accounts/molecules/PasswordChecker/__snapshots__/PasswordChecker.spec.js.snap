// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChecker Snapshot capturing Snapshot of PasswordChecker with different passwords 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-ban fa-w-16 "
    color="red"
    data-icon="ban"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`PasswordChecker Snapshot capturing Snapshot of PasswordChecker with passwords equals 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-check fa-w-16 "
    color="green"
    data-icon="check"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;

exports[`PasswordChecker Snapshot capturing Snapshot of PasswordChecker with too short passwords 1`] = `
<span>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-ban fa-w-16 "
    color="red"
    data-icon="ban"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;
